hack/e2e-suite/guestbook.sh

```diff
@@ -33,25 +33,58 @@ export KUBECTL KUBE_CONFIG_FILE
 source "${KUBE_ROOT}/cluster/kube-env.sh"
 source "${KUBE_VERSION_ROOT}/cluster/${KUBERNETES_PROVIDER}/util.sh"
 
+function wait_for_running() {
+  echo "Waiting for pods to come up."
+  frontends=$(${KUBECTL} get pods -l name=frontend -o template '--template={{range.items}}{{.id}} {{end}}')
+  master=$(${KUBECTL} get pods -l name=redis-master -o template '--template={{range.items}}{{.id}} {{end}}')
+  slaves=$(${KUBECTL} get pods -l name=redisslave -o template '--template={{range.items}}{{.id}} {{end}}')
+  pods=$(echo $frontends $master $slaves)
+
+  all_running=0
+  for i in $(seq 1 30); do
+    sleep 10
+    all_running=1
+    for pod in $pods; do
+      status=$(${KUBECTL} get pods $pod -o template '--template={{.currentState.status}}') || true
+      if [[ "$status" == "Pending" ]]; then
```

Comment:
I am not sure, hence the question mark in my comment. I assume we can get an HTTP GET /watch/<resource-name> ? Perhaps that is not so convenient here. Anyway, this is a very minor point and what you do here is very representative of what we do elsewhere. I'm thinking of changing some of our Go e2e tests that have poll loops with WATCH calls but here I think it is perhaps more convenient to poll and wait.